Userland/Libraries/LibVideo/PlaybackManager.cpp

```diff
@@ -488,20 +536,31 @@ class PlaybackManager::PausedStateHandler : public PlaybackManager::PlaybackStat
     PlaybackState get_state() const override { return PlaybackState::Paused; }
 };
 
+// FIXME: This is a placeholder variable that could be scaled based on how long each frame decode takes to
+//        avoid triggering the timer to check the queue constantly. However, doing so may reduce the speed
+//        of seeking due to the decode thread having to wait for a signal to continue decoding.
+constexpr int buffering_or_seeking_decode_wait_time = 1;
+
 class PlaybackManager::BufferingStateHandler : public PlaybackManager::ResumingStateHandler {
     using PlaybackManager::ResumingStateHandler::ResumingStateHandler;
 
     ErrorOr<void> on_enter() override
     {
-        manager().m_decode_timer->start(0);
+        manager().m_present_timer->start(buffering_or_seeking_decode_wait_time);
         return {};
     }
 
     StringView name() override { return "Buffering"sv; }
 
-    ErrorOr<void> on_buffer_filled() override
+    ErrorOr<void> on_timer_callback() override
```

Comment:
Generally `on_foo` is reserved for `Function<void(Foo&)>` "hooks" that someone wiring up the class will install at top level or in a composition scenario. This one should probably be `timer_did_fire` no? Or whatever the exact pattern is for event handlers. Feels like a "refactor later" comment tho

Reply:
I think this is a fair point, I'll rename it since I already renamed it here anyway :^)